### PR TITLE
fix(doozer): seed lockfile RPMs from latest success or unreleased build

### PR DESCRIPTION
**Pull request by lgarciaaco**
## Summary

Regression follow-up to https://github.com/openshift-eng/art-tools/pull/2851. That change introduced `KonfluxBuildOutcome.UNRELEASED` and persists many successful Konflux builds as `unreleased` (with SBOM-backed `installed_rpms`) before or instead of a `success` row. Lockfile generation still called `get_latest_build(outcome=success)` only, so the **newest** install set could live on `unreleased` while we seeded from an **older** `success` row—stale NVR pins and failing hermetic `dnf` (notably golang-builder).

## Problem

**Before:** `fetch_rpms_from_build` ignored newer `unreleased` DB rows when a `success` row existed with an earlier completion time.

**After:** We load the latest `success` and latest `unreleased`, take the newer by `end_time` (else `start_time`), dedupe identical objects from the two lookups, and if the time-winner has no `installed_rpms`, prefer the other candidate when it does. Unit tests cover ordering and fallback.

## Implementation

- `doozer/doozerlib/image.py`: `_lockfile_seed_build_completion_ts`, `_pick_lockfile_seed_build`, dual `get_latest_build` + info logging in `fetch_rpms_from_build`.
- `doozer/tests/test_image.py`: expect two DB calls where applicable; add tests for newer unreleased vs success and empty-rpms fallback.

No change to `pull_url()` / base-image coordination—this is scoped to lockfile RPM seeding only.

## Validation / test runs

Test Jenkins build (base-image-release): https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Fbase-image-release/5140/

Test Konflux PipelineRun (rhel-9-golang-1-22 / openshift-golang-builder): https://konflux-ui.apps.kflux-ocp-p01.7ayg.p1.openshiftapps.com/ns/ocp-art-tenant/applications/rhel-9-golang-1-22/pipelineruns/rhel-9-golang-1-22-openshift-golang-builder-hprk5

The PLR **builds with hermetic** prefetch as expected; the overall run still **fails at release policy (Conforma / EC)**—that failure is separate from the image build+hermetic path exercised here.

### CODE REVIEW

File: doozer/doozerlib/image.py [hunk 1]
```diff
@@ -28,6 +29,37 @@
...
+    if not candidates:
+        return None
+    # Newer completion time wins; on equal timestamps prefer the latter (UNRELEASED) like strict `>` ordering.
+    chosen = max(
+        enumerate(candidates),
+        key=lambda ib: (_lockfile_seed_build_completion_ts(ib[1]), ib[0]),
+    )[1]
+    if chosen.installed_rpms:
+        return chosen
+    return next((c for c in candidates if c is not chosen and c.installed_rpms), chosen)
```
Comment (locriandev) [inline doozer/doozerlib/image.py]:
```suggestion
  def _pick_lockfile_seed_build(
      success_build: Optional[KonfluxBuildRecord],
      unreleased_build: Optional[KonfluxBuildRecord],
  ) -> Optional[KonfluxBuildRecord]:
      """
      Choose which Konflux DB row should seed lockfile installed_rpms: the latest of SUCCESS and UNRELEASED
      by completion time. If the time-newest row has no installed_rpms, prefer the other when it does.
      """
      candidates = [b for b in (success_build, unreleased_build) if b is not None]
      if not candidates:
          return None

      # Find the build with the latest completion time
      if len(candidates) == 1:
          chosen = candidates[0]
      else:
          # Both success and unreleased exist - compare timestamps
          success_ts = _lockfile_seed_build_completion_ts(candidates[0])
          unreleased_ts = _lockfile_seed_build_completion_ts(candidates[1])

          # Prefer unreleased on tie (>=) to favor newer outcome status
          if unreleased_ts >= success_ts:
              chosen = candidates[1]  # unreleased
          else:
              chosen = candidates[0]  # success

      # If chosen has RPMs, use it; otherwise try the other candidate
      if chosen.installed_rpms:
          return chosen
      return next((c for c in candidates if c is not chosen and c.installed_rpms), chosen)

```

Comment (locriandev) [reply doozer/doozerlib/image.py]: Doesn't change the outcome, just trying to make the function more readable. Having that `max()` comparison with a lambda key operating on tuples is not... ideal for understanding.

Comment (lgarciaaco) [reply doozer/doozerlib/image.py]: refactored the function for better readability [b3c71a7](https://github.com/openshift-eng/art-tools/pull/2876/commits/b3c71a746ea241127c96880060353acdf8ed9a24)

**Review by joepvd — APPROVED** [review]
/lgtm
